packages/components/src/components/hds/application-state/header.hbs

```diff
@@ -3,18 +3,24 @@
   SPDX-License-Identifier: MPL-2.0
 }}
 <div class="hds-application-state__header" ...attributes>
+  {{#if @errorCode}}
+    <Hds::Text::Body
+      class="hds-application-state__error-code"
+      @tag="div"
+      @size="100"
+      @weight="medium"
+      @color="foreground-faint"
+    >
+      Error
+      {{@errorCode}}
+    </Hds::Text::Body>
+  {{/if}}
   {{#if @icon}}
     <div class="hds-application-state__icon">
-      <FlightIcon @name={{@icon}} @size="24" />
+      <FlightIcon @color="var(--token-color-foreground-strong)" @name={{@icon}} @size="24" />
     </div>
   {{/if}}
-  <Hds::Text::Display class="hds-application-state__title" @tag="div" @size="400" @weight="semibold">
+  <Hds::Text::Display class="hds-application-state__title" @tag="div" @size="300" @weight="semibold" @color="strong">
```

Comment:
Thought: This is styled to look like a header visually, so I think it should be a h1-6 component. Looking at Modal and Flyout they also use divs for headers, we don't really have prior examples of how to handle headers in components that could appear anywhere in the page's heading structure.

One way to do this would be to make the tag a sensible default (like h3) and provide an argument for people to change it.

Reply:
I agree. @didoo is there any historical context as to why this is a div? Are you opposed to defaulting to another element?

Reply:
@didoo its fine if we need to leave it as a div... especially for this component where it could be anywhere. For the modal/flyout - because the rest of the page is blocked I think it'd be better to default to h2.

Reply:
I like the idea of adding a `@tag` argument as a pragmatic stop-gap.

Reply:
@didoo Are you suggesting creating a new issue to add the `@tag` argument to the `display` component, or are you saying that users should be able to pass a tag to the `ApplicationState::Header` as part of this work?

Reply:
The `Hds::Text::Display` component [already supports](https://helios.hashicorp.design/components/text#textdisplay) the `@tag` argument, so... the second one 😀

Reply:
Something like `@titleTag`?

Reply:
Yeah, probably that would be the right name to use (there are different elements in the header, and we use `@title` for the text).